support/cas-server-support-saml-idp/src/test/java/org/apereo/cas/support/saml/services/SamlIdPServicesManagerRegisteredServiceLocatorTests.java

```diff
@@ -69,4 +74,37 @@ public void verifyReverseOperation() {
         assertNotNull(result);
         assertTrue(result instanceof SamlRegisteredService);
     }
+
+    /**
+     * serviceLocator should not trigger metadata lookups when requested entityID does not match pattern for service in question.
+     *
+     * This test verifies that, in the case of one service entry that does not match the requested entityID, no
+     * metadata lookups are performed.
+     *
+     * @author Hayden Sartoris
+     */
+    @Test
+    public void verifyEntityIDFilter() {
+        SamlRegisteredServiceCachingMetadataResolver resolver = new BrokenMetadataResolver();
+        SamlIdPServicesManagerRegisteredServiceLocator locator = new SamlIdPServicesManagerRegisteredServiceLocator(resolver);
+
+        val service1 = RegisteredServiceTestUtils.getRegisteredService("urn:abc:def.+");
+        service1.setEvaluationOrder(9);
+        val entityID = "https://sp.testshib.org/shibboleth-sp";
+        val service = webApplicationServiceFactory.createService(entityID);
+
+        locator.locate(List.of(service1), service, r -> r.matches(entityID));
```

Comment:
You must assert the result somehow to actually verify things. The verification might in fact be : "make sure nothing is thrown", etc. if that's what you're going for.